### PR TITLE
`lxd_container`: add check- and diff-mode support

### PR DESCRIPTION
**Pull request by InsanePrawn**
##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

lxd_container

##### ADDITIONAL INFORMATION

Adds check-mode and diff-mode support to the `lxd_container` module.

Cherry-picked and rebased from an older MR of mine.

```diff
--- before
+++ after
@@ -9,7 +9,7 @@
             "image.serial": "20230117_05:24",
             "image.type": "squashfs",
             "image.variant": "cloud",
-            "limits.memory": "3GB"
+            "limits.memory": "30GB"
         },
         "devices": {
             "httpfwd": {

changed: [lxd-test]
```

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Could you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

Also, can you add the following to `DOCUMENTATION`:
```
extends_documentation_fragment:
  - community.general.attributes
attributes:
  check_mode:
    support: full
    version_added: 6.3.0
  diff_mode:
    support: full
    version_added: 6.3.0
```
(potentially with adjusted indentation)

This documents that check and diff mode are supported since community.general 6.3.0.

**Comment by InsanePrawn** [comment]
Thanks for the feedback, done!
I was wondering if the module should also return the "json" dict so it can be used for ingesting instance state into ansible facts? Maybe out of scope for this PR?

File: changelogs/fragments/5866-lxd_container-diff-and-check-mode.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+ - lxd_container - add diff- and check-mode
```
Comment (felixfontein) [inline changelogs/fragments/5866-lxd_container-diff-and-check-mode.yml:2]:
```suggestion
 - lxd_container - add diff and check mode (https://github.com/ansible-collections/community.general/pull/5866).
```

File: plugins/modules/lxd_container.py [hunk 2]
```diff
@@ -742,7 +762,8 @@ def run(self):
             fail_params = {
                 'msg': e.msg,
                 'changed': state_changed,
-                'actions': self.actions
+                'actions': self.actions,
+                'diff': self.diff
```
Comment (felixfontein) [inline plugins/modules/lxd_container.py:766]:
```suggestion
                'diff': self.diff,
```

File: plugins/modules/lxd_container.py [hunk 3]
```diff
@@ -730,7 +749,8 @@ def run(self):
                 'log_verbosity': self.module._verbosity,
                 'changed': state_changed,
                 'old_state': self.old_state,
-                'actions': self.actions
+                'actions': self.actions,
+                'diff': self.diff
```
Comment (felixfontein) [inline plugins/modules/lxd_container.py:753]:
```suggestion
                'diff': self.diff,
```

**Review by felixfontein — COMMENTED** [review]
Just some quick comments. I'll try to give it a more thorough review in the next weeks if none of the module maintainers has time to take a look.

**Comment by InsanePrawn** [comment]
Thanks for the feedback, done!

I suppose I need to change the "added in version" number, now that 6.3 is released? If so, what should I target? 6.4.0?

**Comment by felixfontein** [comment]
Yes, please bump it to 6.4.0. Thanks!

**Comment by felixfontein** [comment]
Looks good enough to me. Since nobody complained, I'll merge.

**Comment by felixfontein** [comment]
@InsanePrawn thanks a lot for your contribution!

**Pull request merged**